notebooks/cuml_benchmarks.ipynb

```diff
@@ -0,0 +1,997 @@
+    ")\n",
+    "\n",
+    "execute_benchmark(\"GaussianRandomProjection\", runner)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Sparse Random Projection"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "runner = cuml.benchmark.runners.SpeedupComparisonRunner(\n",
+    "    bench_rows=[2**x for x in range(17, 24)], \n",
+    "    bench_dims=[64, 128, 256],\n",
+    "    dataset_name=DATA_NEIGHBORHOODS,\n",
+    "    input_type=INPUT_TYPE,\n",
+    "    n_reps=N_REPS\n",
+    ")\n",
+    "\n",
+    "execute_benchmark(\"SparseRandomProjection\", runner)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SVM"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### SVC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "runner = cuml.benchmark.runners.SpeedupComparisonRunner(\n",
+    "    bench_rows=[2**x for x in range(11, 12)], \n",
+    "    bench_dims=[64, 128, 256],\n",
+    "    dataset_name=DATA_CLASSIFICATION,\n",
+    "    input_type=INPUT_TYPE,\n",
+    "    n_reps=N_REPS\n",
+    ")\n",
+    "\n",
+    "execute_benchmark(\"SVC\", runner)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### SVR"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "runner = cuml.benchmark.runners.SpeedupComparisonRunner(\n",
+    "    bench_rows=[2**x for x in range(11, 24)], \n",
+    "    bench_dims=[64, 128, 256],\n",
+    "    dataset_name=DATA_REGRESSION,\n",
+    "    input_type=INPUT_TYPE,\n",
+    "    n_reps=N_REPS\n",
+    ")\n",
+    "\n",
+    "execute_benchmark(\"SVR\", runner)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build Dataframe With Results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.DataFrame(benchmark_results)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def chart_single_algo_speedup(df, algorithm):\n",
+    "    df = df.loc[df.algo == algorithm]\n",
+    "    df = df.pivot(index=\"n_samples\", columns=\"n_features\", values=\"speedup\")\n",
+    "    axes = df.plot.bar(title=\"%s Speedup\" % algorithm)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def chart_all_algo_speedup(df):\n",
+    "    df = df[[\"algo\", \"speedup\"]].groupby([\"algo\"]).mean()\n",
```

Comment:
Should we group by both n_samples and algo? I think that would capture more of the variability of speedups.